Makefile

```diff
@@ -8,6 +8,19 @@ ifndef WORKFLOW_ID
 $(error WORKFLOW_ID variable is not defined. Please provide the required value)
 endif
 
+ifdef APPLICATION_ID
+ifeq ($(APPLICATION_ID), UNDEFINED)
+IS_WORKFLOW = true
+IS_APPLICATION = false
+else
+IS_WORKFLOW = false
+IS_APPLICATION = true
+endif
+else
+IS_WORKFLOW = true
+IS_APPLICATION = false
+endif
+
```

Comment:
Is this a good replacement?
```
# Check if APPLICATION_ID is not defined or set to UNDEFINED
ifeq ($(or $(findstring UNDEFINED,$(APPLICATION_ID)), $(APPLICATION_ID)), UNDEFINED)
IS_WORKFLOW = true
IS_APPLICATION = false
else
IS_WORKFLOW = false
IS_APPLICATION = true
endif
```

Reply:
it's not, when APPLICATION_ID is not defined, the ifeq fails
the most elegant, low-effort alternative that I found is the following (will push commit soon)
```
ifndef APPLICATION_ID
APPLICATION_ID = UNDEFINED
endif

ifeq ($(APPLICATION_ID), UNDEFINED)
IS_WORKFLOW = true
IS_APPLICATION = false
else
IS_WORKFLOW = false
IS_APPLICATION = true
endif
```